### PR TITLE
Dynamic register page

### PR DESCRIPTION
**Pull request by diegoalzate**
## overview
- creates a dynamic register page using new `registration_fields`.
- removes formik and yup and adds react-hook-form. it is a bit more verbose but needed so it can be dynamic
- updates request and response types to api
- validation right now is only checking if required fields are filled

### CODE REVIEW

File: src/api/fetchEvents.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    });
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const res = (await response.json()) as { data: DBEvent[] };
+    return res.data;
+  } catch (error) {
+    console.error('Error fetching nonce:', error);
```
Comment (camilovegag) [inline src/api/fetchEvents.ts:19]: Update console error

File: src/api/fetchRegistration.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ async function fetchRegistration(userId: string): Promise<ResponseProposalType |
       throw new Error(`HTTP error! Status: ${response.status}`);
     }
 
-    const registration = (await response.json()) as { data: ResponseProposalType };
+    const registration = (await response.json()) as { data: GetRegistrationResponseType };
     return registration.data;
   } catch (error) {
     console.error(error);
```
Comment (camilovegag) [inline src/api/fetchRegistration.ts:22]: Maybe add some extra info for this error

File: src/api/fetchRegistrationData.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    );
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const res = (await response.json()) as { data: GetRegistrationDataResponse };
+    return res.data;
+  } catch (error) {
+    console.error('Error fetching nonce:', error);
```
Comment (camilovegag) [inline src/api/fetchRegistrationData.ts:22]: hehe

File: src/api/fetchRegistrationFields.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    );
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const res = (await response.json()) as { data: GetRegistrationFieldsResponse };
+    return res.data;
+  } catch (error) {
+    console.error('Error fetching nonce:', error);
```
Comment (camilovegag) [inline src/api/fetchRegistrationFields.ts:24]: .

File: src/api/postRegistrationFields.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    );
+
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+
+    const res = (await response.json()) as { data: PostRegistrationDataResponse };
+    return res.data;
+  } catch (error) {
+    console.error('Error fetching nonce:', error);
```
Comment (camilovegag) [inline src/api/postRegistrationFields.ts:33]: .

File: src/pages/Register.tsx [hunk 6]
```diff
@@ -1,296 +1,318 @@
...
   const handleSkip = () => {
     setSkipOnboarding('true');
     localStorage.setItem('skip_onboarding', 'true');
   };
 
+  if (isLoading || registrationDataIsLoading) {
+    return <h1>Loading...</h1>;
+  }
+
   if (skipOnboarding == 'true') {
```
Comment (camilovegag) [inline src/pages/Register.tsx]: Change to `===`

File: src/pages/Register.tsx [hunk 7]
```diff
@@ -1,296 +1,318 @@
...
+        {props.options.map((option) => (
+          <option key={option.id} value={option.value}>
+            {option.value}
+          </option>
+        ))}
+      </Select>
+      {props.errors?.[props.id] && <ErrorText>{props.errors?.[props.id]?.message}</ErrorText>}
+    </FlexColumn>
+  );
+}
```
Comment (camilovegag) [inline src/pages/Register.tsx:317]: I think this would be better at /components/form

File: src/types/DBEventType.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+export type DBEvent = {
+  id: string;
+  name: string;
+  createdAt: Date;
```
Comment (camilovegag) [inline src/types/DBEventType.ts:4]: We set some `createdAt` and `updatedAt` dates as strings, shall we keep them here as Dates?

File: src/types/RegistrationDataType.ts [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  id: string;
+  createdAt: Date;
+  updatedAt: Date;
+  registrationId: string;
+  registrationFieldId: string;
+  value: string;
+}[];
+
+export type PostRegistrationDataRequest = {
+  status: 'DRAFT' | 'APPROVED' | 'PUBLISHED' | null;
```
Comment (camilovegag) [inline src/types/RegistrationDataType.ts:11]: status type can be found as `'DRAFT' | 'APPROVED' | 'PUBLISHED' | null;` or `'DRAFT' | 'APPROVED' | 'PUBLISHED' | undefined;`

Will it be good to have some enum type for status only?

**Review by camilovegag — CHANGES_REQUESTED** [review]

File: src/pages/Register.tsx (hunk 7 above)
Comment (diegoalzate) [reply src/pages/Register.tsx:317]: maybe we can do this in another pr. the use case for this field is mainly for registration as it is seen in the errors type

File: src/types/DBEventType.ts (hunk 8 above)
Comment (diegoalzate) [reply src/types/DBEventType.ts:4]: moved it to string

File: src/types/RegistrationDataType.ts (hunk 9 above)
Comment (diegoalzate) [reply src/types/RegistrationDataType.ts:11]: sounds like a good idea. added that

**Review by camilovegag — APPROVED** [review]

**Pull request merged**